infrahub_sdk/spec/object.py

```diff
@@ -198,16 +204,67 @@ def expand_data_with_ranges(data: list[dict[str, Any]]) -> list[dict[str, Any]]:
     return expanded
 
 
+class DataProcessor(ABC):
+    """Abstract base class for data processing strategies"""
+
+    @abstractmethod
+    def process_data(self, data: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        """Process the data according to the strategy"""
+
+
+class SingleDataProcessor(DataProcessor):
+    """Process data without any expansion"""
+
+    def process_data(self, data: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        return data
+
+
+class RangeExpandDataProcessor(DataProcessor):
+    """Process data with range expansion"""
+
+    def process_data(self, data: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        return expand_data_with_ranges(data)
+
+
+class DataProcessorFactory:
+    """Factory to create appropriate data processor based on strategy"""
+
+    _processors: ClassVar[dict[ObjectStrategy, type[DataProcessor]]] = {
+        ObjectStrategy.NORMAL: SingleDataProcessor,
+        ObjectStrategy.RANGE_EXPAND: RangeExpandDataProcessor,
+    }
+
+    @classmethod
+    def get_processor(cls, strategy: ObjectStrategy) -> DataProcessor:
+        processor_class = cls._processors.get(strategy)
+        if not processor_class:
+            raise ValueError(f"Unknown strategy: {strategy}")
```

Comment:
Would be nice to have a test that shows when we get here too.

Hopefully this should be easy to find within an object file. If it's a huge object file it might be nice to have some additional context as to where the event happened. I.e. if you're running `infrahubctl object load` I'm wondering what the error would look like if you have an invalid strategy. Not a requirement for this PR but perhaps you can test for yourself and see what it looks like and determine for yourself. It might be that we have something elsewhere that assists with showing some context that I can't determine from this PR alone.

Reply:
Ive added some better catching of the raised errors: https://github.com/opsmill/infrahub-sdk-python/pull/571/commits/a03008a66aa9e395ed23fc5e5899ec1065e662e1#diff-0da3f5a26233feba6664ea12048ae5aa3c72c548a1bc8b1532225c6021b636a1R241

We will get a `some/path: 1 validation error for InfrahubObjectFileData strategy` if the wrong strategy is used for the enum via pydantic which is good. 

Added a test for that to.